server/data/hmppsRestClient.ts

```diff
@@ -38,36 +40,41 @@ interface StreamRequest {
   errorLogger?: (e: UnsanitisedError) => void
 }
 
-export default class RestClient {
-  agent: Agent
+interface SignedWithMethod {
+  token?: string
+  username?: string
+}
 
-  constructor(private readonly name: string, private readonly config: ApiConfig, private readonly token: string) {
-    this.agent = config.url.startsWith('https') ? new HttpsAgent(config.agent) : new Agent(config.agent)
-  }
+export default class HmppsRestClient {
+  private agent: Agent
 
-  private apiUrl() {
-    return this.config.url
-  }
+  private tokenStore: TokenStore
 
-  private timeoutConfig() {
-    return this.config.timeout
+  constructor(private readonly name: string, private readonly apiConfig: ApiConfig) {
+    this.agent = apiConfig.url.startsWith('https') ? new HttpsAgent(apiConfig.agent) : new Agent(apiConfig.agent)
+    this.tokenStore = new TokenStore()
   }
 
-  async get({ path = null, query = '', headers = {}, responseType = '', raw = false }: GetRequest): Promise<unknown> {
-    logger.info(`Get using admin client credentials: calling ${this.name}: ${path} ${query}`)
+  async get(
+    { path = null, query = {}, headers = {}, responseType = '', raw = false }: GetRequest,
+    signedWithMethod?: SignedWithMethod
```

Comment:
Now introduced this option to the rest client. Can either provide:

1. User token from their session `{ token: 'abcde' }`
2. Username from which to generate a system token associated to the user `{ username: 'joebloggs' }`
3. Supply nothing, in which case a system token will be generated (i.e. community API)